x/supplier/keeper/msg_server_stake_supplier.go

```diff
@@ -44,9 +44,46 @@ func (k msgServer) StakeSupplier(ctx context.Context, msg *types.MsgStakeSupplie
 	if !isSupplierFound {
 		logger.Info(fmt.Sprintf("Supplier not found. Creating new supplier for address %q", msg.Address))
 		supplier = k.createSupplier(ctx, msg)
+
+		// Ensure that only the owner can stake a new supplier.
+		if err = supplier.EnsureOwner(msg.Sender); err != nil {
+			logger.Error(fmt.Sprintf(
+				"owner address %q in the message does not match the msg sender address %q",
+				msg.OwnerAddress,
+				msg.Sender,
+			))
+
+			return nil, err
+		}
+
 		coinsToEscrow = *msg.Stake
 	} else {
 		logger.Info(fmt.Sprintf("Supplier found. About to try updating supplier with address %q", msg.Address))
+
+		// Ensure that only the operator can update the supplier info.
+		if err = supplier.EnsureOperator(msg.Sender); err != nil {
+			logger.Error(fmt.Sprintf(
+				"operator address %q in the message does not match the msg sender address %q",
+				msg.Address,
+				msg.Sender,
+			))
+			return nil, err
```

Comment:
Can you add custom error here that "Ensure that only the operator can update the supplier info"

Reply:
Removed these restrictions, now the owner can update the supplier info too.